pyannote/audio/core/task.py

```diff
@@ -615,47 +618,27 @@ def prepare_data(self):
         self.prepared_data.update(self.prepare_validation())
         self.prepared_data.update(self.post_prepare_data())
 
-        self.has_setup_metadata = True
-
         # save preparated data on the disk
-        if self.cache:
-            with open(self.cache, "wb") as cache_file:
-                np.savez_compressed(cache_file, **self.prepared_data)
+        with open(self.cache, "wb") as cache_file:
+            np.savez_compressed(cache_file, **self.prepared_data)
 
     def post_prepare_data(self):
         return dict()
 
     def setup(self):
-        """Setup data on each device"""
-
-        if self.has_setup_metadata:
-            # if metadata was already generated/assigned to the task, we stop here.
-            # this happens when ...
-
-            return
+        """Setup data cached by prepare_data into the task on each device"""
 
-        if not self.cache:
-            # if no cache directory was provided by the user and task data was not already prepared
-            # this happens when ...
+        fabric = Fabric()
```

Comment:
`fabric` documentation tells me that one should pass a bunch of options to `Fabric` (including the number an type of device).

You do not provide any here: are you sure this works as expected?

Reply:
Feels weird to have to launch a new `fabric` while `Trainer` probably uses its own internally...

Reply:
Currently the task does not have access to the `Trainer` instance (`task.trainer = None`). This is why a new `Fabric ` instance is created here. In principle, the default parameters can be used. But a way to use the `Fabric` instance of the `Trainer` , is to pass by the attribute `trainer` of the `Model`, which need to deplace the line `self.task.model = self` in or to add `self.task.trainer = self.trainer` in the `Model.setup()` , juste before the call to `Task.setup()`

Reply:
> In principle, the default parameters can be used.
Where did you get this information from?

Is there actually a `fabric` attribute in the trainer? 
I can't find any information about this particular aspect in the documentation.